File: 1.1. Prolog/Test-3.pl
:- consult('Task-1.1.pl').
:- initialization(main, main).

main :-
    write('Test 3 (hard): duplicates in first list'), nl,
    remove_twice([1,1,2,3,3], [1,1,3,3], R),
    write('Result:   '), write(R), nl,
    write('Expected: [2]'), nl.

File: 1.1. Haskell/Test-3.hs
import System.IO

countOcc :: Eq a => a -> [a] -> Int
countOcc x = length . filter (== x)

removeIfTwice :: Eq a => [a] -> [a] -> [a]
removeIfTwice list1 list2 = filter (\x -> countOcc x list2 /= 2) list1

main :: IO ()
main = do
    hSetEncoding stdout utf8
    putStrLn "Test 3 (hard): duplicates in first list"
    let l1 = [1,1,2,3,3]
        l2 = [1,1,3,3]
    putStrLn $ "List 1:     " ++ show l1
    putStrLn $ "List 2:     " ++ show l2
    putStrLn $ "Result:     " ++ show (removeIfTwice l1 l2)
    putStrLn   "Expected:   [2]"
